2/homework/laughter_classification/sspnet_data_sampler.py

```diff
@@ -88,14 +108,27 @@ def create_sampled_df(self, frame_sec, naudio=None, save_path=None, force_save=F
         :param force_save: boolean, if you want to override file with same name
         :return:
         """
+        if lazy and (not force_save) and (save_path is not None) and os.path.exists(save_path):
+            return pd.DataFrame.from_csv(save_path, index_col=None)
+
         fullpaths = self.get_valid_wav_paths()[:naudio]
-        dataframes = [self.df_from_file(wav_path, frame_sec) for wav_path in fullpaths]
-        df = pd.concat(dataframes)
+        fullpaths = list(sorted(fullpaths))
 
-        colnames = ["V{}".format(i) for i in range(df.shape[1] - 2)]
-        colnames.append("IS_LAUGHTER")
-        colnames.append("SNAME")
-        df.columns = colnames
+        start_time = time.time()
```

Comment:
You might want to move start_time into `read_dataframe ` funcion fro more consistency
Another way to do this might be using a context manager
https://preshing.com/20110924/timing-your-code-using-pythons-with-statement/

Reply:
`read_dataframe` is a helper for parallel feature computing.
I don't want to move it, because start time means the time before start of parallel computing.